src/passes/OptimizeInstructions.cpp

```diff
@@ -1785,13 +1849,13 @@ struct OptimizeInstructions
     }
     {
       double value;
-      if (matches(curr, binary(Sub, any(), fval(&value))) && value == 0.0) {
-        // x - (-0.0)   ==>   x + 0.0
+      if (matches(curr, binary(Sub, any(), fval(&value)))) {
+        // x - (-C)   ==>   x + C
         if (std::signbit(value)) {
           curr->op = Abstract::getBinary(type, Add);
```

Comment:
Would it make sense to do this even without the sign bit begin set? That is, to do `X - C  =>  X + (-C)`. The binary size is the same, but it would use `+` more often, which could be good for gzip size.

Reply:
Yes, I guess it make sense. LLVM / GCC also do this canonization.